ReactAndroid/src/main/java/com/facebook/react/modules/network/ForwardingCookieHandler.java

```diff
@@ -187,11 +187,7 @@ protected void doInBackgroundGuarded(Void... params) {
         // class. This validates the exception's message to ensure we are only handling this
         // specific exception.
         // https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/webkit/WebViewFactory.java#348
-        if (message != null
-            && exception
-                .getClass()
-                .getCanonicalName()
-                .equals("android.webkit.WebViewFactory.MissingWebViewPackageException")) {
+        if (message != null && message.contains("WebView")) {
```

Comment:
Perhaps the right fix to catch both the original case (from my fix) and the cases by @siddhantsoni, it would be combine both. 

```
if ((message != null && message.contains("No WebView installed")) || 
    exception	
                .getClass()	
                .getCanonicalName()	
                .equals("android.webkit.WebViewFactory.MissingWebViewPackageException")
) {

The reason for this is that if you look at the exceptions listed on #26189, the message does not always contain WebView. Looking back at the crashes listed on #24533, they were also AndroidRuntimeExceptions like the ones that you (@Almouro) are seeing.

Reply:
You are right, your fix catch both cases.

Reply:
In my case the exception message is `android.webkit.WebViewFactory$MissingWebViewPackageException: Failed to load WebView provider: No WebView installed`

So maybe we can check for the message to contain `MissingWebViewPackageException` and that would do it?

Reply:
@Almouro – What do you think about @capezzbr's suggestion? Alternatively would the following sufficiently capture all cases?

```
exception.getClass().getCanonicalName().contains("MissingWebViewPackageException")
```

Reply:
@Almouro, would you be able to take a look at the above comments?

Reply:
Hi @sota000, I created a new PR with the changes requested.

PR: https://github.com/RodolfoGS/react-native/pull/1

Reply:
Hey. It seems that on some devices that crash comes out as a runtime crash "android.util.AndroidRuntimeException" and the check fails. I can reproduce it by disabling the webview on the device. Should we also add it to the check? @RodolfoGS @yungsters @sota000